gspread/exceptions.py

```diff
@@ -61,3 +56,11 @@ def _text_from_detail(
             return dict(errors["error"])
         except (AttributeError, KeyError, ValueError):
             return None
+
+
+class SpreadsheetNotFound(APIError):
+    """Trying to open non-existent or inaccessible spreadsheet."""
+
+    def __init__(self, response: Response):
+        super().__init__(response)
+        self.response: Response = response
```

Comment:
you don't need to set the `self.response` object it's already set by the parent class `APIError`.